pkg/cassandra/config/config.go

```diff
@@ -134,19 +174,46 @@ type SessionBuilder interface {
 	NewSession() (cassandra.Session, error)
 }
 
-// NewSession creates a new Cassandra session
-func (c *Configuration) NewSession() (cassandra.Session, error) {
+// createSession creates session from a configuration
+func createSession(c *Configuration) (cassandra.Session, error) {
 	cluster, err := c.NewCluster()
 	if err != nil {
 		return nil, err
 	}
+
 	session, err := cluster.CreateSession()
 	if err != nil {
 		return nil, err
 	}
+
 	return gocqlw.WrapCQLSession(session), nil
 }
 
+// newSessionPrerequisites creates tables and types before creating a session
+func newSessionPrerequisites(c Configuration) error {
```

Comment:
You are using a very subtle, easy to miss way of passing config by value. I prefer this to be much more explicit:
```suggestion
func (c *Configuration) newSessionPrerequisites(c Configuration) error {
  cfg := *c // clone because we need to connect without specifying a keyspace
  cfg.Schema.Keyspace = ""
```